### ISSUE TITLE
A significant number test files have unrelated accessibility violations

### ISSUE THREAD

**Issue by mraccess77** [act-rules/act-rules.github.io#1165]
Many test pages such as the one video one below have unrelated accessibility violations such as a missing page title.
https://act-rules.github.io/testcases/c5a4ea/26d6803551b1f3b247c4f4750d1291b8d99f5d22.html

Given how test will be run against the unit tests -- It would be helpful if all tests conformed to other rules as well.

**Comment by WilcoFiers** [comment]
@mraccess77 Thank you for raising this issue. I'll put this on the agenda for tomorrow's call.

This is something that has been discussed in the past, but happy to revisit it. The reason we went with this solution in the past is to keep the test cases simple. It's a general principle in software testing to keep tests atomic, not to include more than is strictly necessary for the test you are doing at the time.

I'm still of the opinion this is a good principle to go by for the ACT rules. What I would recommend for implementors is that they only run implementations related to the success criteria that the rule is for. So for example for the Image has accessible name rule, which tests for 1.1.1, only run the rules in your tool related to 1.1.1. This will speed up the test run too.

**Comment by mraccess77** [comment]
Hi @WilcoFiers my line of thinking is so much aimed at implements as it is the average organization trying to compare engines -- perhaps because they want to run in aggregate or have confidence that any engine they select will have similar results.  For these folks testing by success criteria is possible but likely much more work and if they don't do it they are likely to see a lot of issues that they may think are is fails of an engine -- but are actually fails of unrelated tests -- but they likely won't know that.  It just seems odd to have a pass case such as a span with role button link with no accessible name.  If these pass tests end up as examples out of context they could then be wrong in those other contexts.

**Comment by dd8** [comment]
Sometimes the passed tests fail the SC they're supposed to test. For example 'image has accessible name - passed example 8' uses a punctuation smiley as the ALT text for non-text content
https://act-rules.github.io/rules/23a2a8#passed-example-8

The smiley matches the definition of 'non-text content' in SC 1.1.1 so can't be alternative for non-text content:
https://www.w3.org/TR/UNDERSTANDING-WCAG20/text-equiv-all.html#non-text-contentdef

**Comment by Jym77** [comment]
@dd8 That specific test case should probably be updated, I'd say…

---
The problem with having test cases that pass every other rules (or SC) is that they can indeed become pretty big with unrelated information. Part of the unrelated info could be added when we generate the individual files (in the same way that we add `DOCTYPE` and `html` element, we could also add a `lang="en"` attribute and a `title` corresponding to rule+example number). But if there's part that has to be in the example (and not auto-added), I fear we risk making the examples super complicated and harder to read and understand for humans reading the rules as documentation (to the extreme, if I have to dig through 100 lines of boilerplate to find the 1 line relevant to the rule I try to understand, that won't really help me…)

---
Thus said, examples, especially Passed ones, should also be examples of "good practice". Because they will be considered as such be some readers. Having a bad accessible name, or none when needed, is not good practice and these examples should definitely be fixed. Despite the best efforts of rule writers, bad stuff does slip through and we appreciate a lot all the feedback we receive from the community 😀 

---
A related point that always bother me is the compromise we need to make between "all passed and inapplicable examples pass the SC" and "we want examples to have test coverage for all cases in the rule".

This typically appears for inapplicable examples in rules where the applicability is already a bit of testing.

Take for example [HTML page title is descriptive](https://act-rules.github.io/rules/c4a8a4). The Applicability has 3 conditions: 1. there is at least on `title`; 2. it is in a `html`; 3. it has text content.
There is only one inapplicable example which test the second point (the `title` is in a `svg`).
There is no "test coverage" in this rule for the other two points of the applicability. Doing so would require: 1. an `html` element with no `title` descendant; 3. A `title` element with no text content. Both these would fail 2.4.2 which is tested by the rule.

That's typically a case where we have a conflict between "inapplicable examples must pass the SC" and "full test coverage is good". And I'm not sure what to do about it…

**Comment by dd8** [comment]
Yep there's a definite problem with making examples comply with too much. The folks at the PDF Association updating the WCAG PDF Techniques had all sorts of additional best practice requirements on the PDFs examples they produced (many of these were unrelated to accessibility). 

This made producing examples really hard because most PDF producing tools can't produce PDFs that met the requirements.

**Comment by WilcoFiers** [comment]
conclusion: agreed to make the change. We'll take doctype out of the hash, this will change all hashes so far. We'll add a template that will be used when there is no doctype and no HTML element. We'll make these changes when things have stabalized a little more, as to not impact exiting work

**Comment by jkodu** [comment]
Notes from an ACT-R meeting:

- Agreed to make the change. 
- Will take doctype out of the hash, this will change all hashes so far. 
- We'll add a template for examples that will be used when there is no doctype and no HTML element.
- We'll make these changes when things have stabilised a little more, as to not impact exiting work.

**Comment by WilcoFiers** [comment]
These issues have been addressed in the W3C website, we now have a much better template for the examples to be placed into.
